Sources/ApolloCodegenLib/Templates/SelectionSetTemplate.swift

```diff
@@ -396,43 +396,72 @@ fileprivate extension IR.MergedSelections.MergedSource {
   }
 
   private func generatedSelectionSetNameForMergedEntity(in fragment: IR.NamedFragment) -> String {
-    let selectionSetName = ApolloCodegenLib.generatedSelectionSetName(
-      from: typeInfo.scopePath.head.next!,
-      withFieldPath: Array(typeInfo.entity.fieldPath.toArray().dropFirst())
-    )
+    var selectionSetNameComponents: [String] = [fragment.definition.name]
```

Comment:
There is some clean up here as well, but the primary thing happening here is that we now look for type conditions at the root of the fragment and add those to the generated name of the selection set to point to as well. This is what fixes #2220